feat(config): add additional config parameters to Zooz Zen26 and Zen27 and update Zen76/77 parameter language

mattstocum opened the pull request:
Adds missing config parameters to the Zooz Zen26 and Zen27 switches. Breaks out features by firmware version. Updates Zen76 and Zen77 parameter labels and descriptions to match the language used in the Zen26 and Zen27.

AlCalzone commented:
Oh, wow this is a lot. Will review tomorrow 👍🏻

blhoward2 commented on packages/config/config/devices/0x027a/zen26_1.0-1.0.json:
```diff
@@ -12,14 +12,25 @@
+	"associations": {
+		"1": {
+			"label": "Lifeline",
+			"maxNodes": 1,
```
Please remove "supportsZWavePlus": true and the associations groups from all zwave plus devices unless you need a second isLifeline. We're in the process of cleaning up the old files.

blhoward2 replied:
So I get your point with this (and I actually recently opened a bug because we aren't using our name anyway in zwave2jsmqtt), but currently I'm not including associations for zwave plus devices per our earlier conversations. Do you want me to add those back in? I'd assume the ones in the zwa files match the devices but maybe not and we may be able to clean the names up some as we go.

AlCalzone replied:
Let's try without first. If we identify devices that need better labels, we can always add them back.

mattstocum replied:
I'll pull the associations sections out and clear the cache on my dev instance of zwavejs2mqtt later today and see what it comes back with.

mattstocum replied:
I have to go based on a Zen77 right now, because I haven't updated my Zen27 to the firmware that supports group 3 yet due to openzwave issues. The Zen77 reports groups 2 and 3 as "BASIC SET". According to Zooz support, group 3 should be SWITCH MULTILEVEL and send dimming events to associated devices.

blhoward2 replied:
That seems pretty unhelpful and what we want to avoid. Basic Set of what?

AlCalzone replied:
Okay, then leave the section in the config file :)

mattstocum replied:
What's odd, the Zen34 reports Group 3 as "SWITCH_MULTILEVEL CHANGE".

blhoward2 commented on packages/config/config/devices/0x027a/zen26_1.0-1.0.json:
```diff
@@ -50,37 +61,59 @@
+		},
+		"4": {
+			"label": "Auto Turn-Off Timer Duration",
+			"description": "Set the number of seconds to automatically turn off switch once it has been turned on.",
```
This description is unnecessary and can be removed. If the label is clear and there is a unit defined you don't need it.

mattstocum replied:
Excellent! I thought description was a mandatory field for some reason, and a lot of them, my description basically was "does the thing in the label". Description being optional will definitely help make a lot of this cleaner.

blhoward2 replied:
Yeah, that's a recent change. It helps a lot on the ui itself too.

AlCalzone replied:
Actually, that was always optional:
https://zwave-js.github.io/node-zwave-js/#/development/config-files?id=paraminformation
We just had a description for almost every parameter like forever, so it might not seem that way.

blhoward2 commented on packages/config/config/devices/0x027a/zen26_1.0-1.0.json:
```diff
@@ -50,37 +61,59 @@
-			"description": "Set the on off status for the switch after power failure.",
+		"8": {
+			"label": "Switch State After Power Failure",
+			"description": "Set the power state for the switch after power failure.",
```
Unnecessary description. Please remove.

blhoward2 commented on packages/config/config/devices/0x027a/zen26_2.0-2.01.json:
```diff
@@ -0,0 +1,303 @@
+		"2": {
+			"label": "Basic Set",
+			"maxNodes": 5
+		}
```
Ditto above about removing the associations group and the zwave plus field. Please carry throughout.

blhoward2 commented on packages/config/config/devices/0x027a/zen26_2.0-2.01.json:
```diff
@@ -0,0 +1,303 @@
+		},
+		"3": {
+			"label": "Auto Turn-Off Timer",
+			"description": "Enable the auto turn-off timer.",
```
Unnecessary description.

blhoward2 commented on packages/config/config/devices/0x027a/zen26_2.0-2.01.json:
```diff
@@ -0,0 +1,303 @@
+		},
+		"4": {
+			"label": "Auto Turn-Off Timer Duration",
+			"description": "Set the number of minutes to automatically turn off switch once it has been turned on.",
```
Unnecessary description. I'm not going to comment on each, but please take a look throughout.

blhoward2 commented on packages/config/config/devices/0x027a/zen26_2.0-2.01.json:
```diff
@@ -0,0 +1,303 @@
+	],
+	"firmwareVersion": {
+		"min": "2.0",
+		"max": "2.01"
```
Please use the minimum number of digits necessary for the minor number. This should be 2.1 (even though Zooz probably lists 2.01). Please carry throughout.

blhoward2 commented on packages/config/config/devices/0x027a/zen27_1.0-1.0.json:
```diff
@@ -145,72 +135,54 @@
 		"10": {
 			"label": "Minimum Brightness",
-			"description": "Set the minimum brightness level %",
+			"description": "Set the minimum brightness level.",
```
Unnecessary description.

blhoward2 reviewed (changes requested):
Overall these files look great. Please help us clean up these files consistent with what we are doing elsewhere. I've noted what we're trying to clean up in a few places, though I have not flagged each occurrence.

AlCalzone reviewed (changes requested):
Basically what @blhoward2 said. I'm not entirely sure about some of the auto-off timer parameters (see screenshot). Please double-check that they are correct.

mattstocum commented on packages/config/config/devices/0x027a/zen26_2.03-255.255.json:
```diff
@@ -0,0 +1,324 @@
+	],
+	"firmwareVersion": {
+		"min": "2.03",
+		"max": "255.255"
```
Is "zen26_2.03.json" the appropriate name in this case?

AlCalzone replied:
without the 0, yes.

mattstocum commented on packages/config/config/devices/0x027a/zen26_1.0-1.0.json:
```diff
@@ -50,37 +61,59 @@
-			"description": "enable the Auto Turn-Off Timer",
-			"valueSize": 2,
+			"description": "Enable the auto turn-off timer.",
+			"valueSize": 1,
 			"minValue": 0,
-			"maxValue": 32767,
+			"maxValue": 1,
```
I wasn't sure for version 1.0. For the later firmware revisions, they changed the unit to minutes and increased the max to 65535 and increased the field size to 4 bytes. https://www.support.getzooz.com/kb/article/315-zen26-s2-on-off-switch-ver-3-02-advanced-settings/ For version 1.0 I wasn't sure, but I'm going to assume the manual is correct since it also uses seconds as the units. I'll fix this, and also double check the valueSize on the rest of the files.

AlCalzone replied:
A little background: Z-Wave config parameters are meant to be signed by default (unless a device interprets them differently, which is why we have the `unsigned` flag). So you'd have a range of -32768..+32767. I've seen enough files where +32768 (or similar for 1 und 4-byte values) was used on accident, so I'm careful around this stuff.

mattstocum replied:
Actually, after reading the manual, I'm even more confused. The manual says the timer is controlled entirely in parameter 3, but the docs have parameter 3 as enabled/disabled, and parameter 4 as the length. I'm going to reach out to Zooz support on this one to see if they still have documentation for version 1.0 of the firmware available.

mattstocum replied:
I'm still waiting to hear back from Zooz support, but for now, I'm going to go under the assumption that the manuals for the 26 and 27 correctly describe the config parameters for the 1.0 firmware for each device. The copyright year on the manuals do seem to align with the firmware release date. For newer firmware revisions, I'm trusting the documentation on their website or my own testing where I can test.

mattstocum commented:
I think I addressed everything above. I also updated the Zen34 config to match the guidance from the above comments, and added parameter 7 to the Zen76 and Zen77. The online docs don't list it as supported, but I checked with Zooz support, and they confirmed it is supported, and I was able to test it myself to verify.

AlCalzone commented on packages/config/config/devices/0x027a/zen27_2.8.json:
```diff
@@ -24,13 +24,15 @@
 		"2": {
```
Can't select the correct line, but the firmwareVersion.min should be `2.8`.

AlCalzone replied:
btw, the lint is failing because of this but your branch is out of date with master, so we don't see it for some reason.

mattstocum replied:
> A few nits. Also: Do you know how the association labels look like for the ones where you removed the association config? Having something like `BASIC_SET` is not so nice.

Basically everything is "BASIC_SET". What would be the preferred language? Group 2 sends on/off commands only, group 3, on the devices that support it, sends dimming events. The labels I had were just "Basic Set" and "Switch Multilevel". Maybe just "On/Off" and "Dimmer" or "On/Off Control" and "Dimmer Control"?

> btw, the lint is failing because of this but your branch is out of date with master, so we don't see it for some reason.

Yeah, I messed up when I made the branch and am not confident enough with git to fix things. I was hoping since I was only changing files that weren't changed in master that it wouldn't be an issue. If it is, I think I can stash my changes, re-create the branch off the current master, and then create a new PR.

AlCalzone replied:
> The labels I had were just "Basic Set" and "Switch Multilevel". Maybe just "On/Off" and "Dimmer" or "On/Off Control" and "Dimmer Control"?

I'm fine with all of them. @blhoward2 what is most consistent with your changes?

AlCalzone replied:
> If it is, I think I can stash my changes, re-create the branch off the current master, and then create a new PR.

Replace `upstream` with name my remote has:
```
git fetch upstream
git rebase upstream/master
```
If there are conflicts, fix them and check them in, then
```
git rebase --continue
```
and repeat.

In the end if all went well:
```
git push --force
```

blhoward2 replied:
> > The labels I had were just "Basic Set" and "Switch Multilevel". Maybe just "On/Off" and "Dimmer" or "On/Off Control" and "Dimmer Control"?
> 
> 
> 
> I'm fine with all of them. @blhoward2 what is most consistent with your changes?

I'd say 'Basic Set - On/Off' and 'Send Dim Level' to make it clear what they do.

AlCalzone commented on packages/config/config/devices/0x027a/zen77.json:
```diff
@@ -30,11 +26,13 @@
 		}
 	},
-	"supportsZWavePlus": true,
+	"firmwareVersion": {
+		"min": "0.0",
+		"max": "255.255"
+	},
```
this should be above associations (for consistency)

AlCalzone reviewed (changes requested):
A few nits. Also: Do you know how the association labels look like for the ones where you removed the association config? Having something like `BASIC_SET` is not so nice.

AlCalzone commented:
@blhoward2 anything else you need changed?

blhoward2 commented:
Nope

AlCalzone commented:
thanks @mattstocum

bepsoccer commented:
I will work on updating some of the Zooz configs as I updated many of the OZW config files for them on behalf of Zooz a while back and I can say at a minimum the Zen26 and Zen27 updates here aren't entirely accurate.  For example the firmware should be listed as 2.03, not 2.3 as that is how Zooz themselves refer to it.  Also, I'm not sure where those parameters came from with this merge, but here's is the last update I made for the Zen26 in OZW; https://github.com/OpenZWave/open-zwave/blob/master/config/zooz/zen26.xml.

Any how, I'm excited to hopefully start moving my 150+ devices to ZwaveJS in the coming months and will do what ever I can to help contribute where I can.

mattstocum commented:
Z-Wave JS removes the leading zeros from the version, so 2.3 is the correct value for the firmware version. With regards to the additional parameters for the Zen26 and 27 https://www.support.getzooz.com/kb/article/315-zen26-s2-on-off-switch-ver-3-02-advanced-settings/ and https://www.support.getzooz.com/kb/article/314-zen27-s2-dimmer-ver-3-01-advanced-settings/. I was planning on starting to work on additional Zooz devices and add the metadata for inclusion/exclusion, etc, to the devices I've already added. In the interest of not duplicating work, if you let me know which ones you were going to work on, I'll take them off my list.
On 2/9/2021 9:51:12 PM, bepsoccer <notifications@github.com> wrote:
I will work on updating some of the Zooz configs as I updated many of the OZW config files for them on behalf of Zooz a while back and I can say at a minimum the Zen26 and Zen27 updates here aren't entirely accurate. For example the firmware should be listed as 2.03, not 2.3 as that is how Zooz themselves refer to it. Also, I'm not sure where those parameters came from with this merge, but here's is the last update I made for the Zen26 in OZW; https://github.com/OpenZWave/open-zwave/blob/master/config/zooz/zen26.xml [https://github.com/OpenZWave/open-zwave/blob/master/config/zooz/zen26.xml].
Any how, I'm excited to hopefully start moving my 150+ devices to ZwaveJS in the coming months and will do what ever I can to help contribute where I can.
—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub [https://github.com/zwave-js/node-zwave-js/pull/1564#issuecomment-776401032], or unsubscribe [https://github.com/notifications/unsubscribe-auth/AAADDGTM42NNEAHXQNAJ2TDS6HYCBANCNFSM4W7Q2Y6A].

blhoward2 commented:
Before you spend time adding the metadata, check whether the Zooz device is in the zwave alliance database. If it is, we'll parse those into the file automatically.

mattstocum commented:
Some of them are, but what's in the alliance DB is pretty bad for Zooz. Does the automatic merge handle devices with multiple firmware revisions, like the Zen26/27?
On 2/10/2021 8:59:56 AM, blhoward2 <notifications@github.com> wrote:
Before you spend time adding the metadata, check whether the Zooz device is in the zwave alliance database. If it is, we'll parse those into the file automatically.
—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub [https://github.com/zwave-js/node-zwave-js/pull/1564#issuecomment-776723859], or unsubscribe [https://github.com/notifications/unsubscribe-auth/AAADDGU6EE7GDM4M4WAXLEDS6KGNVANCNFSM4W7Q2Y6A].

blhoward2 commented:
Yes to the extent they're there.
